src/main/java/com/google/devtools/build/lib/skyframe/RepositoryMappingFunction.java

```diff
@@ -178,7 +179,7 @@ public SkyValue compute(SkyKey skyKey, Environment env)
       return computeFromWorkspace(repositoryName, externalPackageValue, rootModuleRepoMapping);
     }
 
-    throw new RepositoryMappingFunctionException();
+    return RepositoryMappingValue.NOT_FOUND_VALUE;
```

Comment:
I couldn't for the life of me get this thing to work with `env.getValueOrThrow` 😅 For some reason, I can't use `getOrThrow` to "swallow" the exception. It seems that Skyframe always expects me to massage the exception and throw something else, instead of allowing me to say "it's fine, I got it handled"; the exception always ends up bubbling up. Here's a failed attempt if you're curious: https://gist.github.com/Wyverald/bdca2d5b1b1657db683b7f61c1ffedd1

Reply:
> Would this impact error reporting in other cases where this isn't expected?

But also, just to this point -- `RepositoryMappingValue.key()` has never been used with a nonexistent repo up until now. It was only ever used by something within an existing repo, to figure out the mappings for that repo.

Reply:
Thanks for trying, it wasn't clear to me whether `env.getValueOrThrow` was just meant as a way to improve error details during bubbling or as some kind of "let me try something else" mechanism.

Reply:
Confirmed with @michaeledgar; once a SkyFunction throws, its node is marked as failed. So we aren't really meant to "recover" from this.